package/crds/projects.gitlab.crossplane.io_projects.yaml

```diff
@@ -162,8 +162,30 @@ spec:
                       to be true.
                     type: integer
                   importUrl:
-                    description: URL to import repository from.
-                    type: string
+                    description: URL to import repository from. Provided credentials
+                      in the URL will be overwritten in case a valid secretRef is
+                      present in ImportUrlSecretRef.
+                    minLength: 11
+                    type: string
+                  importUrlSecretRef:
+                    description: Secret to use when importing project with secret.
+                      Provided credentials in ImportUrl will be overwitten with the
+                      credentials found in ImportUrlSecretRel.
+                    properties:
+                      password:
+                        type: string
+                      secretName:
+                        type: string
+                      secretNamespace:
+                        type: string
+                      username:
+                        type: string
+                    required:
+                    - password
+                    - secretName
+                    - secretNamespace
+                    - username
```

Comment:
Are `username` and `password` required values or should the provider support imports from public repos?

Reply:
username and password are required only if `importUrlSecretRef` is not nil. And yes the provider should also support imports from public repos. @abacus3

Reply:
OK, thank. I think the name `importUrlSecretRef´ is a bit misleading. Can we change it to something more explicit like `importCredentialsSecretRef`?